pandas/core/generic.py

```diff
@@ -1948,6 +1948,129 @@ def tail(self, n=5):
             return self
         return self.iloc[-n:]
 
+  
+    def sample(self, n=None, frac=None, replace=False, weights=None, random_state=None, axis = 0):
+        """
+        Returns a random sample of rows from object. 
+        
+        Parameters
+        ----------
+        n : int, optional
+            Number of rows to return. Cannot be used with `frac`.
+            Default = 1 if `frac` = None. 
+        frac : float, optional
+            Share of rows to return. Cannot be used with `n`. 
+        replace : boolean, optional
+            Sample with or without replacement. Default = False. 
+        weights : str or ndarray-like, optional
+            Default 'None' results in equal probability weighting. 
+            If called on a DataFrame or Panel, will also accept the name of a 
+            column as a string. Must be same length as index. 
+            If weights do not sum to 1, they will be normalized to sum to 1. 
+            Missing values in the weights column will be treated as zero. 
+            inf and -inf values not allowed. 
+        random_state : int or numpy.random.RandomState, optional
+            Seed for the random number generator (if int), or numpy RandomState 
+            object.
+        axis : int, optional
+            Axis to sample. Default = 0.
+            
+        Returns
+        -------
+        Same type as caller. 
+        """
+        
+        ###        
+        # Processing axis argument
+        ###
+        
+        # This checks validity of axis argument.
+        if axis > (self.ndim - 1):
+            raise ValueError("Axis argument out of range")
```

Comment:
use ``axis = self._get_axis_number(axis)`` will raise (and translate such things as ``axis='columns'`` to the correct axis number